src/App.tsx

```diff
@@ -15,36 +17,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+interface Rules {
+  sortField: SortField | '';
+  isReversed: boolean;
+}
+
+function sortGoods(array: string[], { sortField, isReversed }: Rules) {
+  const result = [...array];
+
+  switch (sortField) {
+    case SortField.alphabet:
+      result.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortField.length:
+      result.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    result.reverse();
+  }
+
+  return result;
+}
+
+export const App = () => {
+  const [objOfRules, setObjOfRules] = useState<Rules>({
+    sortField: '',
+    isReversed: false,
+  });
+  let goods = [...goodsFromServer];
+
+  goods = sortGoods(goods, objOfRules);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn({
+            button: true,
+            'is-info': true,
+            'is-light': objOfRules.sortField !== SortField.alphabet,
+          })}
+          onClick={() =>
+            setObjOfRules({
+              ...objOfRules,
+              sortField: SortField.alphabet,
+            })
+          }
+        >
           Sort alphabetically
         </button>
-
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn({
+            button: true,
+            'is-success': true,
+            'is-light': objOfRules.sortField !== SortField.length,
+          })}
+          onClick={() =>
+            setObjOfRules({
+              ...objOfRules,
+              sortField: SortField.length,
+            })
+          }
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)